app/assets/javascripts/ransack_ui_jquery/index.js

```diff
@@ -1,2 +1,6 @@
 //= require ransack/predicates
-//= require ransack_ui_jquery/search_form
\ No newline at end of file
+//= require ransack_ui_jquery/search_form
+
+if(jQuery.fn.select2) {
+  window.Select2 = jQuery.fn.select2;
+}
```

Comment:
Select2 v4 = `jQuery.fn.select2`
Select2 v3 = `window.Select2`
Originally we we're checking for (window) `Select2`, hence the revert